packages/notifier/src/notifier.js

```diff
@@ -75,11 +75,13 @@ export const makeNotifierKit = (...args) => {
         // If hasState() and either it is final() or it is
         // not the state of updateCount, return the current state.
         assert(currentResponse !== undefined);
-        return Promise.resolve(currentResponse);
```

Comment:
This change is responsible for divergent behaviour between local notifiers and remote notifiers, as detected in the wallet tests.

Reply:
Restored. Resolving.